File: leetcode/string_compression/string_compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {

        vector<char> ans_chars;
        char now=chars[0];
        int length=0;
        for(int i=0;i<chars.size();i++){
            if(chars[i]==now){
                length++;
            }else{
                ans_chars.push_back(now);
                if(length==1){
                    length=1;
                    now=chars[i];
                    continue;
                }
                vector<char> temp;
                while(length/10!=0){
                    int decamical=length%10;
                    temp.push_back(decamical + '0');
                    length=length/10;
                }
                temp.push_back(length + '0');

                for(int j=temp.size()-1;j>=0;j--){
                    ans_chars.push_back(temp[j]);
                }
                length=1;
                now=chars[i];
            }
        }
        ans_chars.push_back(now);
        if(length!=1){
            vector<char> temp;
            while(length/10!=0){
                int decamical=length%10;
                temp.push_back(decamical + '0');
                length=length/10;
            }
            temp.push_back(length + '0');

            for(int j=temp.size()-1;j>=0;j--){
                ans_chars.push_back(temp[j]);
            }
        }
        int ans = ans_chars.size();

        chars=ans_chars;
        
        return ans;
    }
};

File: leetcode/string_compression/string_compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        write=0
        curr_char=chars[0]
        curr_num=0

        for c in chars:
            if curr_char==c:
                curr_num+=1
            else:
                chars[write]=curr_char
                write+=1
                curr_char=c
                if curr_num>1:
                    for d in str(curr_num):
                        chars[write]=d
                        write+=1
                curr_num=1
                
        chars[write]=curr_char
        write+=1
        if curr_num>1:
            for d in str(curr_num):
                chars[write]=d
                write+=1
        
        return write